File: DB/Frontend/src/types/culturalItem.ts
import { CulturalCategoryId } from './category';

export interface ImageMeta {
  url: string;
  alt: string;
  caption?: string;
  credit: string;
  license: string;
  source?: string;
}

export interface SourceCitation {
  title: string;
  url?: string;
  publisher: string;
  verifiedDate: string;
}

export interface StarSchemaNode {
  id: string;
  type: 'history' | 'region' | 'materials' | 'tradition' | 'festival' | 'modern' | 'people' | 'music' | 'technique' | 'tribal' | 'fashion' | 'art' | 'architecture';
  label: string;
  shortDescription: string;
  detailedContent: string;
}

export interface RecipeInfo {
  prepTime?: string;
  cookTime?: string;
  difficulty?: 'Easy' | 'Medium' | 'Advanced';
  ingredientsSummary: string[];
  culturalContext: string;
  recipeUrl: string;
  verifiedSourceName: string;
}

export interface CulturalItem {
  id: string;
  slug: string;
  title: string;
  marathiTitle?: string;
  hindiTitle?: string;
  stateId: string;
  category: CulturalCategoryId;
  shortDescription: string;
  description: string;
  history: string;
  culturalSignificance: string;
  location: {
    name: string;
    district: string;
    state: string;
    coordinates: {
      lat: number;
      lng: number;
    };
  };
  images: ImageMeta[];
  primaryImage: string;
  sources: SourceCitation[];
  starSchemaNodes: StarSchemaNode[];
  relatedItemSlugs: string[];
  tags: string[];
  recipeInfo?: RecipeInfo;
  model3DId?: string;
  lastVerified: string;
}

File: DB/Backend/database/models/cultural_item.py
"""
Pydantic models for CulturalItem data.
Mirrors Frontend/src/types/culturalItem.ts — CulturalItem and nested types.
"""

from typing import Optional, List, Any
from pydantic import BaseModel


class ImageMetaResponse(BaseModel):
    url: str
    alt: str = ""
    caption: Optional[str] = None
    credit: str = ""
    license: str = ""
    source: Optional[str] = None


class SourceCitationResponse(BaseModel):
    title: str
    url: Optional[str] = None
    publisher: str = ""
    verifiedDate: str = ""


class StarSchemaNodeResponse(BaseModel):
    id: str
    type: str  # history/region/materials/tradition/festival/modern/etc
    label: str = ""
    shortDescription: str = ""
    detailedContent: str = ""


class RecipeInfoResponse(BaseModel):
    prepTime: Optional[str] = None
    cookTime: Optional[str] = None
    difficulty: Optional[str] = None  # Easy/Medium/Advanced
    ingredientsSummary: List[str] = []
    culturalContext: str = ""
    recipeUrl: str = ""
    verifiedSourceName: str = ""


class LocationResponse(BaseModel):
    name: str = ""
    district: str = ""
    state: str = ""
    coordinates: dict = {}  # {lat: float, lng: float}


class CulturalItemResponse(BaseModel):
    """Full CulturalItem — matches Frontend CulturalItem interface."""
    id: str
    slug: str
    title: str
    marathiTitle: Optional[str] = None
    hindiTitle: Optional[str] = None
    stateId: str = ""
    category: str = ""
    shortDescription: str = ""
    description: str = ""
    history: str = ""
    culturalSignificance: str = ""
    location: LocationResponse = LocationResponse()
    images: List[ImageMetaResponse] = []
    primaryImage: str = ""
    sources: List[SourceCitationResponse] = []
    starSchemaNodes: List[StarSchemaNodeResponse] = []
    relatedItemSlugs: List[str] = []
    tags: List[str] = []
    recipeInfo: Optional[RecipeInfoResponse] = None
    model3DId: Optional[str] = None
    lastVerified: str = ""


class CulturalItemSummaryResponse(BaseModel):
    """Lightweight summary for list endpoints."""
    id: str
    slug: str
    title: str
    stateId: str = ""
    category: str = ""
    shortDescription: str = ""
    primaryImage: str = ""
    location: LocationResponse = LocationResponse()
    tags: List[str] = []
